pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -820,6 +823,27 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 	}, nil
 }
 
+func (og *operationGenerator) GenerateVerifyControllerDetachedVolumeFunc(
+	volumeToDetach AttachedVolume,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+	return func() error {
+		if !volumeToDetach.PluginIsAttachable {
+			// If the volume does not implement the attacher interface, it is
+			// assumed to be detached and the actual state of the world is
+			// updated accordingly.
+			actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+			return nil
+		}
+
+		glog.Infof("Controller detached volume %q (spec.Name: %q) devicePath: %q",
+			volumeToDetach.VolumeName,
+			volumeToDetach.VolumeSpec.Name(),
+			volumeToDetach.DevicePath)
+		actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+		return nil
```

Comment:
I think we have to put the logic of checking whether volume is globally mounted or not inside of this function. Otherwise there might be a race condition. When reconciler checks globalMounted is false, and then right after that, this state is changed to true. When calling VerifyControllerDetachVolume it will think it is not mounted and mark it as detached.